numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -507,6 +507,73 @@ string_expandtabs_loop(PyArrayMethod_Context *context,
 }
 
 
+template <ENCODING enc>
+static int
+string_center_ljust_rjust_loop(PyArrayMethod_Context *context,
+        char *const data[], npy_intp const dimensions[],
+        npy_intp const strides[], NpyAuxData *NPY_UNUSED(auxdata))
+{
+    JUSTPOSITION pos = *(JUSTPOSITION *)(context->method->static_data);
+    int elsize1 = context->descriptors[0]->elsize;
+    int elsize3 = context->descriptors[2]->elsize;
+    int outsize = context->descriptors[3]->elsize;
+
+    char *in1 = data[0];
+    char *in2 = data[1];
+    char *in3 = data[2];
+    char *out = data[3];
+
+    npy_intp N = dimensions[0];
+
+    while (N--) {
+        Buffer<enc> buf(in1, elsize1);
+        Buffer<enc> fill(in3, elsize3);
+        Buffer<enc> outbuf(out, outsize);
+        size_t len = string_pad(buf, *(npy_int64 *)in2, fill, pos, outbuf);
+        if (len < 0) {
+            return -1;
+        }
+
+        in1 += strides[0];
+        in2 += strides[1];
+        in3 += strides[2];
+        out += strides[3];
+    }
+
+    return 0;
+}
+
+
+template <ENCODING enc>
+static int
+string_zfill_loop(PyArrayMethod_Context *context,
+        char *const data[], npy_intp const dimensions[],
+        npy_intp const strides[], NpyAuxData *NPY_UNUSED(auxdata))
+{
+    int elsize = context->descriptors[0]->elsize;
+    int outsize = context->descriptors[2]->elsize;
+
+    char *in1 = data[0];
+    char *in2 = data[1];
+    char *out = data[2];
+
+    npy_intp N = dimensions[0];
+
+    while (N--) {
+        Buffer<enc> buf(in1, elsize);
+        Buffer<enc> outbuf(out, outsize);
+        npy_intp newlen = string_zfill(buf, *(npy_int64 *)in2, outbuf);
```

Comment:
If you have the error check for the others, probably good to have it here too `if (newlen < 0)`